core/uint128.go

```diff
@@ -0,0 +1,78 @@
+package core
```

Comment:
Let's move this to it's own package. We typically reserve the `core` package for Starknet-related types.

Following [`uint256`](https://github.com/holiman/uint256)'s lead, the package name can be `uint128` and the struct name can be `Int` to avoid stutters.